sig-scalability/goals.md

```diff
@@ -0,0 +1,78 @@
+# Kubernetes Scaling and Performance Goals
+
+_by Quinton Hoole and Wojciech Tyczynski, Google Inc_
+
+**April 2016**
+
+This document is a markdown version converted from a working [Google Doc](https://docs.google.com/document/d/1pABjn69yeDJDl0Yrsneiwj5A_bUI7BZgvKTQtSwzjiE/edit).  Please refer to the original for extended commentary and discussion.
+
+## Introduction
+
+What size clusters do we think that we should support with Kubernetes in the short to medium term?  How performant do we think that the control system should be at scale?  What resource overhead should the Kubernetes control system reasonably consume?  This document attempts to answer these and related questions.
+
+## Background
+Based on our experience running very large production systems at Google, we have found that running excessively large compute and/or storage clusters has several downsides, most notably:
+
+1. **Large-scale cluster failures can lead to large losses of capacity**: Despite our best efforts, correlated cluster-wide hardware or software failures do happen occasionally.  Even with built-in redundancy, load shedding, and good system design practises, the reality is that data center power supplies, and their backup systems do sometimes fail at the same time. Software rollouts, reconfigurations, system overloads and other cascading failures do occasionally go wrong at cluster-wide scale.  When they do (not if they do), it's better not to lose all (or even half of) your total capacity.
+2. **Scale for the sake of scale can lead to unnecessarily complex system designs**:    Many relatively simple system designs have large, but not infinite, natural scaling limits, and in order to go beyond those limits, more complex designs become necessary.  Unless the need for the additional scale can be justified, the additional complexity often cannot be.
+
+Where Google's internal requirements for compute capacity exceed the natural scaling limits of a cluster (or even multiple clusters, required for redundancy or geo-locality), we typically choose to build more clusters, rather than trying to build larger clusters.  While the resultant increase in the number of clusters can be seen as an unnecessary administrative burden, we have found that the benefits of this approach far outweigh this disadvantage. In addition we have built internal automated tooling which helps us to more easily manage larger numbers of clusters, and accommodate applications that span multiple clusters.
```

Comment:
I think this should be more of a best practices / lessons for certain use cases.  Not everyone can afford the many clusters approach, or they may have specialized requirements.  As a community we want to address those use cases where it's feasible.